source/MaterialXView/Viewer.cpp

```diff
@@ -1308,25 +1308,56 @@ void Viewer::loadDocument(const mx::FilePath& filename, mx::DocumentPtr librarie
                     // Generate a shader for the new material.
                     mat->generateShader(_genContext);
                 }
-
-                mx::NodePtr materialNode = mat->getMaterialNode();
-                if (materialNode)
+            }
+       
+            // For document assignment we must go in order of assignments for looks
+            // with later assignments superceding earlier ones.
+            for (mx::LookPtr look : doc->getLooks())
```

Comment:
Previous logic was to traverse through materials found in the document and then find the match in the look assignments in order. This can give incorrect results. Instead traverse by look first and match with materials found. This allows for the "last match" to set the final material assignment.